File: src/main/java/com/lessmarkup/dataobjects/migrations/Initial.java
package com.lessmarkup.dataobjects.migrations;

import com.lessmarkup.dataobjects.*;
import com.lessmarkup.interfaces.data.Migration;
import com.lessmarkup.interfaces.data.Migrator;

public class Initial implements Migration {
    @Override
    public String getId() {
        return "20150410_210800";
    }

    @Override
    public void migrate(Migrator migrator) {
        migrator.createTable(User.class);
        migrator.createTable(Currency.class);
        migrator.createTable(EntityChangeHistory.class);
        migrator.createTable(AttachedFile.class);
        migrator.createTable(Image.class);
        migrator.createTable(Language.class);
        migrator.createTable(Menu.class);
        migrator.createTable(Module.class);
        migrator.createTable(NodeAccess.class);
        migrator.createTable(Node.class);
        migrator.createTable(NodeUserData.class);
        migrator.createTable(SiteCustomization.class);
        migrator.createTable(Smile.class);
        migrator.createTable(SuccessfulLoginHistory.class);
        migrator.createTable(TestMail.class);
        migrator.createTable(Translation.class);
        migrator.createTable(UserBlockHistory.class);
        migrator.createTable(UserGroupMembership.class);
        migrator.createTable(UserGroup.class);
        migrator.createTable(UserLoginIpAddress.class);
        migrator.createTable(UserPropertyDefinition.class);
        migrator.createTable(SiteProperties.class);
        migrator.addDependency(Translation.class, Language.class);
        migrator.addDependency(NodeUserData.class, Node.class);
        migrator.addDependency(NodeUserData.class, User.class);
        migrator.addDependency(NodeAccess.class, Node.class);
        migrator.addDependency(UserBlockHistory.class, User.class);
        migrator.addDependency(UserBlockHistory.class, User.class, "BlockedByUserId");
        migrator.addDependency(UserGroupMembership.class, User.class);
        migrator.addDependency(UserGroupMembership.class, UserGroup.class);
        migrator.addDependency(UserLoginIpAddress.class, User.class);
    }
}

File: src/main/scala/com/lessmarkup/dataobjects/migrations/Initial.scala
package com.lessmarkup.dataobjects.migrations

import com.lessmarkup.dataobjects._
import com.lessmarkup.interfaces.data.{Migration, Migrator}

class Initial extends Migration {
  def getId: String = {
    "20150410_210800"
  }

  def migrate(migrator: Migrator) {
    migrator.createTable(classOf[User])
    migrator.createTable(classOf[Currency])
    migrator.createTable(classOf[EntityChangeHistory])
    migrator.createTable(classOf[AttachedFile])
    migrator.createTable(classOf[Image])
    migrator.createTable(classOf[Language])
    migrator.createTable(classOf[Menu])
    migrator.createTable(classOf[Module])
    migrator.createTable(classOf[NodeAccess])
    migrator.createTable(classOf[Node])
    migrator.createTable(classOf[NodeUserData])
    migrator.createTable(classOf[SiteCustomization])
    migrator.createTable(classOf[Smile])
    migrator.createTable(classOf[SuccessfulLoginHistory])
    migrator.createTable(classOf[TestMail])
    migrator.createTable(classOf[Translation])
    migrator.createTable(classOf[UserBlockHistory])
    migrator.createTable(classOf[UserGroupMembership])
    migrator.createTable(classOf[UserGroup])
    migrator.createTable(classOf[UserLoginIpAddress])
    migrator.createTable(classOf[UserPropertyDefinition])
    migrator.createTable(classOf[SiteProperties])
    migrator.addDependency(classOf[Translation], classOf[Language])
    migrator.addDependency(classOf[NodeUserData], classOf[Node])
    migrator.addDependency(classOf[NodeUserData], classOf[User])
    migrator.addDependency(classOf[NodeAccess], classOf[Node])
    migrator.addDependency(classOf[UserBlockHistory], classOf[User])
    migrator.addDependency(classOf[UserBlockHistory], classOf[User], "blockedByUserId")
    migrator.addDependency(classOf[UserGroupMembership], classOf[User])
    migrator.addDependency(classOf[UserGroupMembership], classOf[UserGroup])
    migrator.addDependency(classOf[UserLoginIpAddress], classOf[User])
  }
}
